packages/x-components/src/x-modules/url/store/actions/update-url.action.ts

```diff
@@ -0,0 +1,102 @@
+import { MappedParam, MappedParams, UrlXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link UrlActions.updateUrl}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @public
+ */
+export const updateUrl: UrlXStoreModule['actions']['updateUrl'] = ({
+  getters: {
+    urlParams: { page, ...params }
+  }
+}) => {
+  const currentUrl = window.location.href;
+  const url = new URL(currentUrl);
+
+  clearUrlParams(url, params);
+  urlSetPage(url, page);
+  urlSetParams(url, params);
+
+  window.history.replaceState({ ...window.history.state }, document.title, url.href);
+};
+
+/**
+ * Clears the parameters in the URL.
+ *
+ * @param url - The current URL.
+ * @param mappedParams - The params in {@link UrlXStoreModule.state} mapped with the overridden keys
+ * if they are.
+ *
+ * @public
```

Comment:
These module scoped functions shouldn't be public, they should be either private or internal